docker-compose-dev.yaml

```diff
@@ -0,0 +1,120 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+version: '2.1'
+
+services:
+  processor:
+    build:
+      context: .
+      dockerfile: ./processor/Dockerfile-dev
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    image: marketplace-tp-dev
+    volumes:
+      - '.:/project/sawtooth-marketplace'
+    command: echo "No transaction processor command set"
+
+  server:
+    build:
+      context: .
+      dockerfile: ./server/Dockerfile-dev
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    image: marketplace-server-dev
+    volumes:
+      - .:/project/sawtooth-marketplace
+    ports:
+      - '8000:8000'
+    depends_on:
+      - ledger-sync
+      - rethink
+      - validator
+    command: echo "No server command set"
+
+  ledger-sync:
+    build:
+      context: .
+      dockerfile: ./ledger_sync/Dockerfile-dev
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    image: marketplace-ledger-sync-dev
+    volumes:
+      - .:/project/sawtooth-marketplace
+    depends_on:
+      - rethink
+      - validator
+    command: echo "No ledger sync command set"
+
+  rethink:
+    image: rethinkdb:2.3
+    ports:
+      - '9090:8080'
+    expose:
+      - 8080
+      - 28015
+
+  rest-api:
+    image: hyperledger/sawtooth-rest-api:0.8
+    container_name: sawtooth-rest-api
+    expose:
+      - 8008
+    ports:
+      - '8008:8008'
+    depends_on:
+      - validator
+    entrypoint: sawtooth-rest-api --connect tcp://validator:4004 --bind rest-api:8008 -vv
+
+  shell:
+    image: hyperledger/sawtooth-shell:0.8
+    volumes:
+      - '.:/project/sawtooth-marketplace'
+    working_dir: /project/sawtooth-marketplace
+    container_name: sawtooth-shell
+    depends_on:
+      - rest-api
+    entrypoint: |
+      bash -c "
+        sawtooth keygen &&
+        tail -f /dev/null
```

Comment:
isn't `sleep infinity` a better choice?

Reply:
The /dev/null approach is more portable and is a very common approach.  Not sure if sleep infinity works in busybox-based images, for example.

Reply:
When do we build busybox  based images?